lib/elixir/src/elixir_module.erl

```diff
@@ -226,11 +231,12 @@ build(Line, File, Module, Lexical) ->
 
 eval_form(Line, Module, Data, Block, Vars, E) ->
   {Value, EE} = elixir_compiler:eval_forms(Block, Vars, E),
-  elixir_overridable:store_pending(Module),
+  OverridableDefinitions = elixir_overridable:store_pending(Module),
   EV = elixir_env:linify({Line, reset_env(EE)}),
   EC = eval_callbacks(Line, Data, before_compile, [EV], EV),
-  elixir_overridable:store_pending(Module),
-  {Value, EC}.
+  OverriddenDefinitions = elixir_overridable:store_pending(Module),
+  OverridablePairsNotOverridden = OverridableDefinitions -- OverriddenDefinitions,
```

Comment:
@samphilipd I am absolutely sure. :) What happens in this case:

```elixir
defmodule BeforeCompile do
  defmacro __before_compile__(_) do
    quote do
      def bar, do: :bar
      defoverridable bar: 0
    end
  end
end

defmodule Sample do
  @before_compile BeforeCompile
  def foo, do: :foo
  defoverridable foo: 0
end
```

One should return `[foo: 0]` and the other `[bar: 0]`.